schemas/debug-adapters.schema.json

```
@@ -11,36 +11,173 @@
-        "gdbserver":  { "type": "null", "description": "Add the gdbserver: node in the cbuild-run.yml file." },
-        "defaults":   { "$ref": "#/definitions/DebugAdaptersDefaultsType" }
+        "name": {
+          "type": "string",
+          "description": "Debugger adapter name."
+        },
+        "alias-name": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "List of names (in input node or BSP) that map to this debug adapter."
+        },
+        "template": {
+          "type": "string",
+          "description": "Used only by the VS Code CMSIS Solution extension for configuration."
+        },
+        "gdbserver": {
+          "type": "null",
+          "description": "Add the gdbserver: node in the cbuild-run.yml file."
+        },
+        "defaults": {
+          "$ref": "#/definitions/DebugAdaptersDefaultsType"
+        },
+        "ui": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/UserInterfaceType"
+          }
+        }
       },
       "additionalProperties": false,
-      "required": ["name"]
+      "required": [
+        "name"
+      ]
     },
     "DebugAdaptersDefaultsType": {
       "description": "List of default options to use when not specified in the target-set: node.",
       "type": "object",
       "properties": {
-        "port":     { "type": "number", "description": "Port number." },
-        "protocol": { "enum": [ "jtag", "swd" ], "description": "Selected debug protocol (jtag or swd)." },
-        "clock":    { "type": "number", "description": "Selected debug clock speed (in Hz)." }
+        "port": {
+          "type": "number",
+          "description": "Port number."
+        },
+        "protocol": {
+          "enum": [
+            "jtag",
+            "swd"
+          ],
+          "description": "Selected debug protocol (jtag or swd)."
+        },
+        "clock": {
+          "type": "number",
+          "description": "Selected debug clock speed (in Hz)."
+        }
       },
       "additionalProperties": true
+    },
+    "UserInterfaceType": {
+      "description": "User interface configuration.",
+      "type": "object",
+      "properties": {
+        "section": {
+          "type": "string",
+          "description": "Section name for the user interface."
+        },
+        "description": {
+          "type": "string",
+          "description": "Description of the section."
+        },
+        "yml-node": {
+          "type": "string",
+          "description": "YAML node in the cbuild-run.yml file."
+        },
+        "type": {
+          "enum": [
+            "number",
+            "enum",
+            "string"
+          ],
+          "description": "Type of the user interface element."
+        },
+        "select": {
+          "enum": [
+            "on",
+            "off"
+          ],
```

This is a [boolean](https://json-schema.org/understanding-json-schema/reference/boolean), `on`/`off` should be recognized as boolean by yml parsers compatible with yml spec 1.1 (ref https://yaml.org/type/bool.html).
```suggestion
          "type": "boolean",
```